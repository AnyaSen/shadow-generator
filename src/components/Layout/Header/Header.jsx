import React from "react";

import Styles from "./Header.module.scss";

export default function Header() {
  return (
    <div className={Styles.Header}>
      <h1>SHADOW GENERATOR</h1>
    </div>
  );
}
