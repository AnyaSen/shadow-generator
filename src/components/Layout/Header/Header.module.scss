@import "../../../assets/scss/variables";

.Header {
  padding-top: $space-3;
  padding-left: $space-3;
  color: $color-dark-tapestry;
  width: fit-content;
}
