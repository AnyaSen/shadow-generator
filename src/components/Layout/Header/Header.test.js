import React from "react";
import { render, cleanup } from "@testing-library/react";
import Header from "./Header";
import { BrowserRouter } from "react-router-dom";

const createRenderTree = () => (
  <BrowserRouter>
    <Header />
  </BrowserRouter>
);

describe("<Header />", () => {
  afterEach(cleanup);

  test("Should contain the correct text", () => {
    const headingText = "SHADOW GENERATOR";

    const tree = createRenderTree();

    const { getByText, container } = render(tree);

    const heading = getByText(headingText);

    expect(heading).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
