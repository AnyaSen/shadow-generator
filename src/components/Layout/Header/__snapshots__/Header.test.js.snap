// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should contain the correct text 1`] = `
<div
  class="Header"
>
  <a
    href="/"
  >
    <h1>
      SHADOW GENERATOR
    </h1>
  </a>
</div>
`;
